st2common/st2common/util/output_schema.py

```diff
@@ -69,8 +70,9 @@ def mask_secret_output(ac_ex, output_value):
         return output_value
 
     for key, spec in output_schema.items():
-        if key in output_value[output_key] and spec.get("secret", False):
-            output_value[output_key][key] = MASKED_ATTRIBUTE_VALUE
+        if isinstance(output_value[output_key], Collection):
```

Comment:
Hmm. On closer inspection, maybe this check should move up a few lines (lines 69-70) so that we don't have to check it for every entry in the output_schema.

```python
    if not output_value.get(output_key) or not isinstance(output_value[output_key], Collection):
        return output_value
```

And while you're in there, we could probably simplify the dict/list check on lines 60-61:
```python
    if not isinstance(output_value, Collection):
        return output_value
```

Reply:
Sorry for changing my suggestion. Thanks again!

Reply:
I don't want to change anymore than I have.  It sounds like you have some idea of how to clean this up more so go ahead and take over if you would like.